compiler/rustc_const_eval/src/util/compare_types.rs

```diff
@@ -59,19 +60,23 @@ pub fn relate_types<'tcx>(
         Err(_) => return false,
     };
     let errors = ocx.select_all_or_error();
-    // With `Reveal::All`, opaque types get normalized away, with `Reveal::UserFacing`
-    // we would get unification errors because we're unable to look into opaque types,
-    // even if they're constrained in our current function.
-    for (key, ty) in infcx.take_opaque_types() {
-        let hidden_ty = tcx.type_of(key.def_id).instantiate(tcx, key.args);
-        if hidden_ty != ty.hidden_type.ty {
-            span_bug!(
-                ty.hidden_type.span,
-                "{}, {}",
-                tcx.type_of(key.def_id).instantiate(tcx, key.args),
-                ty.hidden_type.ty
-            );
+
+    if anchor != DefiningAnchor::Error {
+        // With `Reveal::All`, opaque types get normalized away, with `Reveal::UserFacing`
```

Comment:
The relationship between what the `if` is checking (`DefiningAnchor`, whatever that is) and what the comment talks about (`Reveal::UserFacing`/`Reveal::All`) is completely unclear.

Reply:
Well I didn't touch that comment, but I'll rework it I guess.